src/basic_hierarchy/interfaces/Instance.java

```diff
@@ -1,9 +1,30 @@
 package basic_hierarchy.interfaces;
 
-public interface Instance {
+/**
+ * An instance is a singular entity that can be placed in the broader context of a {@link Hierarchy}.
+ */
+public interface Instance
+{
+	/**
+	 * @return the name of this particular instance
+	 */
 	public String getInstanceName();
-	public String getNodeId();
+
+	/**
+	 * @return data columns associated with this instance
+	 */
 	public double[] getData();
+
+	/**
+	 * @return id of the true class (group) to which this instance has been assigned ('ground truth').
```

Comment:
@return id of the true class (group) from which this instance comes from ('ground truth').